### PR TITLE
Feature/contributing docs

### PR DESCRIPTION
**Pull request by GeorgeCiesinski**
Created a `CONTRIBUTING.md`, `CODE_OF_CONDUCT.md` and updated `README.md`.

### CODE REVIEW

**Comment by GeorgeCiesinski** [comment]
@Wissididom when you have some time, can you take a look at the added files to see if you can find any issues that need to be updated? If needed I will update my PR before we merge.

File: docs/CODE_OF_CONDUCT.md [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+an individual is officially representing the community in public spaces.
+Examples of representing our community include using an official e-mail address,
+posting via an official social media account, or acting as an appointed
+representative at an online or offline event.
+
+## Enforcement
+
+Instances of abusive, harassing, or otherwise unacceptable behavior may be
+reported to the community leaders responsible for enforcement at
+our [discord](https://discord.gg/WmD5CW3aFu).
```
Comment (Wissididom) [inline docs/CODE_OF_CONDUCT.md:63]: I'm not sure if we should capitalize the name `Discord` here
```suggestion
our [Discord](https://discord.gg/WmD5CW3aFu).
```

File: docs/CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+First off, thanks for taking the time to contribute! ❤️
+
+All types of contributions are encouraged and valued. See the [Table of Contents](#table-of-contents) for different ways to help and details about how this project handles them. Please make sure to read the relevant section before making your contribution. It will make it a lot easier for us maintainers and smooth out the experience for all involved. The community looks forward to your contributions. 🎉
+
+> And if you like the project, but just don't have time to contribute, that's fine. There are other easy ways to support the project and show your appreciation, which we would also be very happy about:
+> - Star the project
+> - Tweet about it
+> - Refer this project in your project's readme
+> - Mention the project at local meetups and tell your friends/colleagues
+> - Join our [discord](#community)
```
Comment (Wissididom) [inline docs/CONTRIBUTING.md:12]: Not sure if we should capitalize Discord here too.
```suggestion
> - Join our [Discord](#community)
```

File: docs/CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+    - [Your First Code Contribution](#your-first-code-contribution)
+    - [Improving The Documentation](#improving-the-documentation)
+  - [Styleguides](#styleguides)
+    - [Commit Messages](#commit-messages)
+  - [Attribution](#attribution)
+
+
+## Community
+
+If you are interested in this project, please consider joining our [discord](https://discord.gg/WmD5CW3aFu).
```
Comment (Wissididom) [inline docs/CONTRIBUTING.md:36]: I'm not sure if we should capitalize the name `Discord` here too
```suggestion
If you are interested in this project, please consider joining our [Discord](https://discord.gg/WmD5CW3aFu).
```

File: docs/CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+> If you want to ask a question, we assume that you have read the available [Wiki](https://github.com/GeorgeCiesinski/poke-guesser-bot/wiki).
+
+Before you ask a question, it is best to search for existing [Issues](https://github.com/GeorgeCiesinski/poke-guesser-bot/issues) that might help you. In case you have found a suitable issue and still need clarification, you can write your question in this issue. It is also advisable to search the internet for answers first.
+
+If you then still feel the need to ask a question and need clarification, we recommend the following:
+
+- Open an [Issue](https://github.com/GeorgeCiesinski/poke-guesser-bot/issues/new).
+- Provide as much context as you can about what you're running into.
+- Provide project and platform versions (nodejs, npm, etc), depending on what seems relevant.
+- Label the issue `question`.
```
Comment (Wissididom) [inline docs/CONTRIBUTING.md:50]: Afaik you can not label PR's yourself if you are neither Owner, Member or Collaborator, but I could be wrong. Not sure if we have a template where we could I think set it to automatically assign a label on creation.

File: docs/CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+
+We will then take care of the issue as soon as possible.
+
+> Alternatively, you can join our [discord](#community) and ask your question there.
```
Comment (Wissididom) [inline docs/CONTRIBUTING.md:54]: Not sure if we should capitalize Discord here too.
```suggestion
> Alternatively, you can join our [Discord](#community) and ask your question there.
```

File: docs/CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+> When contributing to this project, you must agree that you have authored 100% of the content, that you have the necessary rights to the content and that the content you contribute may be provided under the project license.
+
+### Reporting Bugs
+
+#### Before Submitting a Bug Report
+
+A good bug report shouldn't leave others needing to chase you up for more information. Therefore, we ask you to investigate carefully, collect information and describe the issue in detail in your report. Please complete the following steps in advance to help us fix any potential bug as fast as possible.
+
+- Make sure that you are using the latest version.
+- Determine if your bug is really a bug and not an error on your side e.g. using incompatible environment components/versions (Make sure that you have read the [wiki](https://github.com/GeorgeCiesinski/poke-guesser-bot/wiki). If you are looking for support, you might want to check [this section](#i-have-a-question)).
```
Comment (Wissididom) [inline docs/CONTRIBUTING.md:69]: Maybe capitalize the `Wiki` here:
```suggestion
- Determine if your bug is really a bug and not an error on your side e.g. using incompatible environment components/versions (Make sure that you have read the [Wiki](https://github.com/GeorgeCiesinski/poke-guesser-bot/wiki). If you are looking for support, you might want to check [this section](#i-have-a-question)).
```

File: docs/CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,162 @@
...
+
+### Suggesting Enhancements
+
+This section guides you through submitting an enhancement suggestion for Poke-guesser Bot, **including completely new features and minor improvements to existing functionality**. Following these guidelines will help maintainers and the community to understand your suggestion and find related suggestions.
+
+
+#### Before Submitting an Enhancement
+
+- Make sure that you are using the latest version.
+- Read the [wiki](https://github.com/GeorgeCiesinski/poke-guesser-bot/wiki) carefully and find out if the functionality is already covered, maybe by an individual configuration.
```
Comment (Wissididom) [inline docs/CONTRIBUTING.md:106]: Maybe capitalize the `Wiki` here too:
```suggestion
- Read the [Wiki](https://github.com/GeorgeCiesinski/poke-guesser-bot/wiki) carefully and find out if the functionality is already covered, maybe by an individual configuration.
```

File: docs/CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,162 @@
...
+- Perform a [search](https://github.com/GeorgeCiesinski/poke-guesser-bot/issues) to see if the enhancement has already been suggested. If it has, add a comment to the existing issue instead of opening a new one.
+- Find out whether your idea fits with the scope and aims of the project. It's up to you to make a strong case to convince the project's developers of the merits of this feature. Keep in mind that we want features that will be useful to the majority of our users and not just a small subset. If you're just targeting a minority of users, consider creating a fork.
+
+
+#### How Do I Submit a Good Enhancement Suggestion?
+
+Enhancement suggestions are tracked as [GitHub issues](https://github.com/GeorgeCiesinski/poke-guesser-bot/issues).
```
Comment (Wissididom) [inline docs/CONTRIBUTING.md:113]:
```suggestion
Enhancement suggestions are tracked as [GitHub Issues](https://github.com/GeorgeCiesinski/poke-guesser-bot/issues).
```

**Review by Wissididom — COMMENTED** [review]
I just looked through it and mentions a few things I felt like being better, I'm open for discussing them.
The git flow page I'd need to checkout myself because I did not use the naming convention for branches and maybe find out a few more things through that.

File: docs/CONTRIBUTING.md (hunk 4 above)
Comment (GeorgeCiesinski) [reply docs/CONTRIBUTING.md]: Good point. I removed that point from the document.

**Comment by GeorgeCiesinski** [comment]
Regarding the git-flow, no worries. It isn't really a standard but a suggestion on how to use github to organize branches. It helps later when it comes to managing existing and outdated branches, and it makes the git log graph much easier to understand visually.

File: docs/CONTRIBUTING.md [hunk 9]
```diff
@@ -0,0 +1,161 @@
...
+- Collect information about the bug:
+- Stack trace (Traceback) if applicable
+- OS, Platform and Version (Windows, Linux, macOS, x86, ARM)
+- Version of the interpreter, compiler, SDK, runtime environment, package manager, depending on what seems relevant.
+- Steps to reproduce the bug
+
+
+#### How Do I Submit a Good Bug Report?
+
+> For security related issues, please report a security vulnerability [here](https://github.com/GeorgeCiesinski/poke-guesser-bot/security/advisories/new).
```
Comment (Wissididom) [inline docs/CONTRIBUTING.md:80]: You'd need to create one for that link to work, I think:
image

When you go on https://github.com/GeorgeCiesinski/poke-guesser-bot/security/advisories it seems like that there are the settings for it.

Comment (GeorgeCiesinski) [reply docs/CONTRIBUTING.md:80]: I was trying to link to this issue type. Do you know if contributors have access to this? Do you have access?

Screenshot 2023-02-26 at 3 38 52 PM

Comment (GeorgeCiesinski) [reply docs/CONTRIBUTING.md:80]: This is in the issues tab: https://github.com/GeorgeCiesinski/poke-guesser-bot/issues

Comment (GeorgeCiesinski) [reply docs/CONTRIBUTING.md:80]: I also double checked and it looks like it is for publicly reporting security issues, so it should work: 

https://docs.github.com/en/code-security/security-advisories/guidance-on-reporting-and-writing/privately-reporting-a-security-vulnerability

Comment (Wissididom) [reply docs/CONTRIBUTING.md:80]: I first checked the advisories link than I got this:
image
Then I clicked on New on Issues and got:
image
After that I've then closed the tab and reopened it with Ctrl+Shift+T and now I see the third option:
image

So TL;DR I seem to see it now, don't know if non Collaborators see it.

Comment (GeorgeCiesinski) [reply docs/CONTRIBUTING.md:80]: I think I fixed it, which is why it worked for you. I had to enable a beta option in the settings for it. Okay then in that case I will update it.

Comment (Wissididom) [reply docs/CONTRIBUTING.md:80]: Jip, now the link with the new in the end seems to work fine, I don't get a 404 anymore

**Comment by GeorgeCiesinski** [comment]
I'm going to go ahead and merge this in. 

Regarding minor points like the git-flow question and styling guide, we can always modify that on the go. Maybe we can talk about it in the meeting.

**Pull request merged**